### ISSUE TITLE
destDir resolving to root and ../ not supported

### ISSUE THREAD

**Issue by brunoimbrizi** [sapphi-red/vite-plugin-static-copy#46]
I bumped into a small bug and an unsupported case when using `flatten: false`

The small bug: `destDir` resolves to the root folder when `dest` is empty ([utils.ts line 66](https://github.com/sapphi-red/vite-plugin-static-copy/blob/main/src/utils.ts#L66))

This example results in the contents of /locales copied to C:/ (assuming Windows):
```
viteStaticCopy({ targets: [ { src: 'locales/**/*.json', dest: '' } ], flatten: false })
```
---

The unsupported case: parent folders `../` as sources

This example works as expected and copies /locales to the destination:
```
viteStaticCopy({ targets: [ { src: '../../static/locales', dest: '' } ] })
```

This example copies /static to the destination (which is C:/ on Windows, as mentioned above):
```
viteStaticCopy({ targets: [ { src: '../../static/locales/**/*.json', dest: '' } ], flatten: false })
```

**Comment by FluffyDiscord** [comment]
Also issue on Linux. Can this be fixed please?

**Comment by brunoimbrizi** [comment]
After working on #57 and getting more familiar with the tests I can expand on the issue with two examples.
Assume you add the following to [vite.noflatten.config.ts](https://github.com/sapphi-red/vite-plugin-static-copy/blob/main/test/fixtures/vite.noflatten.config.ts):

_Example 1_
```
viteStaticCopy({ targets: [ { src: '../../test/fixtures2/*.txt', dest: 'fixture3' } ], flatten: false })
```
In the current version, [destDir](https://github.com/sapphi-red/vite-plugin-static-copy/blob/main/src/utils.ts#L74) resolves to `fixture3/../test/fixtures2` which outputs the file to:
```
dist-noflatten/
└── test/
    └── fixtures2/
        └── baz.txt
```
The expected destination would've been `dist-noflatten/fixture3/test/fixtures2/baz.txt`.


_Example 2_
```
viteStaticCopy({ targets: [ { src: '../fixtures2/*.txt', dest: '' } ], flatten: false })
```
In this case, [destDir](https://github.com/sapphi-red/vite-plugin-static-copy/blob/main/src/utils.ts#L74) resolves to `/fixtures2` which outputs the file to:
```
C:/
└── fixtures2/
    └── baz.txt
```
The expected destination would've been `dist-noflatten/fixtures2/baz.txt`.

**Comment by sapphi-red** [comment]
fixed by #57
